src/contract/contract.rs

```diff
@@ -185,39 +186,100 @@ impl<State: ExposedState> OutputAssignment<State> {
     }
 }
 
+#[derive(Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Hash, Debug, Display)]
+#[derive(StrictType, StrictDumb, StrictEncode, StrictDecode)]
+#[strict_type(lib = LIB_NAME_RGB, dumb = { Self(1) })]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate", rename_all = "camelCase")
+)]
+#[display(inner)]
+pub struct WitnessHeight(u32);
+
+impl WitnessHeight {
+    pub fn new(height: u32) -> Option<Self> {
+        match height {
+            0 => None,
+            height => Some(WitnessHeight(height)),
+        }
+    }
+
+    pub fn get(&self) -> NonZeroU32 { NonZeroU32::new(self.0).expect("invariant") }
+}
+
+/// RGB consensus information about the current mined height of a witness
+/// transaction defining the ordering of the contract state data.
+#[derive(Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Hash, Debug, Display)]
+#[derive(StrictType, StrictDumb, StrictEncode, StrictDecode)]
+#[strict_type(lib = LIB_NAME_RGB, tags = order)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate", rename_all = "camelCase")
+)]
+pub enum WitnessOrd {
+    #[display(inner)]
+    OnChain(WitnessHeight),
+
+    #[display("offchain")]
+    #[strict_type(dumb)]
+    OffChain,
+}
+
+impl WitnessOrd {
+    pub fn from_electrum_height(height: u32) -> Self {
```

Comment:
As we intend to support other explorers, like esplora, I think it's interesting to change this name.

```suggestion
    pub fn from_block_height(height: u32) -> Self {
```

Reply:
Well, it's not block height, it is just an Electrum API which reports 0 for mempool (Bitcoin Core doesn't do that; esplora inherits electrum API). The name MUST indicate that this is something very custom and not a part of anything standard (like block height)

Reply:
It could be something like `from_explorer_height` ?

Reply:
done